fe/fe-core/src/main/java/com/starrocks/catalog/HivePartitionKey.java

```diff
@@ -29,4 +31,34 @@ public HivePartitionKey() {
     public List<String> nullPartitionValueList() {
         return ImmutableList.of(HiveMetaClient.PARTITION_NULL_VALUE);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        PartitionKey that = (PartitionKey) o;
+
+        if (getKeys().size() != that.getKeys().size()) {
+            return false;
+        }
+
+        for (int i = 0; i < getKeys().size(); i++) {
+            LiteralExpr literalExpr = getKeys().get(i);
+            LiteralExpr thatLiteralExpr = that.getKeys().get(i);
+
+            // like hour=06 and hour=6, they are two partitions in hms
+            if (literalExpr instanceof IntLiteral && thatLiteralExpr instanceof IntLiteral) {
```

Comment:
if 06 is int literal, it is the same as intliteral.
Because the values of IntLiteral are the same

Reply:
The core problem is IntLiteral. Before this pr, we only compared the numbers inside. `06` and `6` were both 6, so the `equals()` will return true. But in Hive, `06` and `6` are two partitions, even if they are all `Int` type. So we need to handle `IntLiteral` `equals()` behavior specially in HivePartitionKey.